components/Form/Button.tsx

```diff
@@ -1,4 +1,12 @@
-export const Button = ({ loading, children, type, ...props }) => (
+import { FC } from 'react';
+
+type Props = {
+  loading?: boolean;
+  children: string;
+  type: 'button' | 'submit' | 'reset' | undefined;
```

Comment:
Usually, when you use a spread pattern in HTML element you should extend your custom props via union type.

When you hover on the element you will see what type it has.
<img width="332" alt="CleanShot 2022-06-22 at 21 34 01@2x" src="https://user-images.githubusercontent.com/12488648/175121281-b522e339-d837-4430-8661-080996d2c51d.png">

You can copy that, and use it as 

```
type Props = {
  loading?: boolean;
  children: string;
  type: 'button' | 'submit' | 'reset' | undefined;
} & JSX.IntrinsicElements.button

```

That will take care of the native props like `type`

Reply:
If you want to avoid some props, use only particular or make them optional.

Check typescript's Pick or Omit